catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveCommand.java

```diff
@@ -84,19 +88,15 @@ private Object executeRemoveFromStore() throws Exception {
         CatalogFacade catalogProvider = getCatalog();
 
         if (hasFilter()) {
-            QueryImpl query = new QueryImpl(getFilter());
-
-            query.setRequestsTotalResultsCount(true);
-            query.setPageSize(-1);
-
-            Map<String, Serializable> properties = new HashMap<>();
-            properties.put("mode", "native");
-
-            SourceResponse queryResponse = catalogProvider.query(new QueryRequestImpl(query,
-                    properties));
-
-            final List<String> idsFromFilteredQuery = queryResponse.getResults()
-                    .stream()
+            ResultIterable resultIterable = new ResultIterable(catalogFramework,
+                    new QueryRequestImpl(new QueryImpl(getFilter(),
+                            1,
+                            PAGE_SIZE,
+                            SortBy.NATURAL_ORDER,
+                            true,
+                            TimeUnit.SECONDS.toMillis(30)), false));
```

Comment:
Perhaps extract timeout value to constant higher up in class so that it is easier to change in the future.